Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseObjectCollection.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Collections
+{
+    public abstract class BaseObjectCollection : MonoBehaviour
+    {
+        #region public members
+        /// <summary>
+        /// Action called when collection is updated
+        /// </summary>
+        public Action<BaseObjectCollection> OnCollectionUpdated;
```

Comment:
Needs to be a property if public

Reply:
Where are we even creating and assigning this action?

Reply:
in the UpdateCollection() method. This action gets called every time a user presses the custom inspector button. Like I said earlier, I haven't actually ever subscribed to this action in practice.